library/math/complex.lisp

```diff
@@ -24,8 +26,21 @@
 #+coalton-release
 (cl:declaim #.coalton-impl/settings:*coalton-optimize-library*)
 
+;;; Tell Coalton that we should expect Complex to have a special type representation.
+(cl:eval-when (:compile-toplevel :load-toplevel :execute)
+  (cl:setf (cl:gethash 'Complex coalton-impl/typechecker/lisp-type:*special-type-representations*)
+           (cl:lambda (ty)
+             ;; (cl:format cl:t "testing ~S~%" ty)
+             (cl:typecase ty
+               (ty:tycon
+                (cl:case (ty:tycon-name ty)
+                  (coalton:single-float (cl:values '(cl:complex cl:single-float) cl:t))
+                  (coalton:double-float (cl:values '(cl:complex cl:double-float) cl:t))
+                  (cl:otherwise (cl:values '(cl:or cl:number Complex) cl:t))))
+               (cl:t
+                (cl:values '(cl:or cl:number Complex) cl:t))))))
```

Comment:
It should be `(cl:or cl:number cl:complex)` instead of `(cl:or cl:number Complex)` (?)

Reply:
It's [this complex](https://github.com/coalton-lang/coalton/blob/main/library/math/complex.lisp#L29)

Reply:
Oh, okay! `Complex` denotes the complex structure-class that gets defined with the coalton type.